server/routerlicious/packages/routerlicious-base/src/utils/sessionHelper.ts

```diff
@@ -287,7 +288,35 @@ export async function getSession(
 ): Promise<ISession> {
 	const baseLumberjackProperties = getLumberBaseProperties(documentId, tenantId);
 
-	const document = await documentRepository.readOne({ tenantId, documentId });
+	let document;
+	try {
+		// Retry document existence check to avoid document DB race condition
+		document = await runWithRetry(
+			async () =>
+				documentRepository.readOne({ tenantId, documentId }).then((result) => {
+					if (result === null) {
+						throw new NetworkError(
+							404,
+							"Document is deleted and cannot be accessed",
+							true /* canRetry */,
+						);
+					}
+					return result;
+				}),
+			"getDocumentForSession",
+			2, // maxRetries
+			500, // retryAfterMs
```

Comment:
This will make all real 404s take a minimum of `1.5s`. Instead, could we switch this to 50ms which will bring that down to 150ms?

Otherwise, we will need to make sure the right people are informed about why our getSession latency metrics are terrible. We get millions of (presumably) legitimate 404s every day with P95 of <100ms, so this will absolutely tank our perf metrics.

Reply:
Given the perf implications here, we should also consider making this retryAfterMs configurable for easier fine-tuning.

Reply:
Decreased retry time and made it configurable